hal/hal_uv.c

```diff
@@ -32,12 +32,8 @@ static adb_context_uv_t g_adb_context;
 adb_context_t* adb_hal_create_context() {
     adb_context_uv_t *adbd = &g_adb_context;
 
-#ifdef __NUTTX__
-    uv_library_init(&adbd->uv_context);
-    adbd->loop = uv_default_loop(&adbd->uv_context);
-#else
```

Comment:
By default libuv uses static memory for the default loop. As NuttX uses flat memory for all the processes, this default loop is shared so it's not possible to use libuv in multiple processes.

That's why the uv library port on NuttX is updated a little to allow multiple processes to use it. This is configured with the `CONFIG_LIBUV_CONTEXT` flag, which is missing here, causing the compilation error.

I would suggest to use this although I'm not sure the `CONFIG_LIBUV_CONTEXT` option is forwarded properly. What do you think?
```
#if defined(__NUTTX__) && defined(CONFIG_LIBUV_CONTEXT)
```

Reply:
@spiriou multiple libuv instance is finally supported by NuttX now, all hack for NuttX can be removed from microADB, please review the change https://github.com/spiriou/microADB/pull/23.